pgx/src/datum/interval.rs

```diff
@@ -0,0 +1,189 @@
+/*
+Portions Copyright 2019-2021 ZomboDB, LLC.
+Portions Copyright 2021-2022 Technology Concepts & Design, Inc. <support@tcdi.com>
+
+All rights reserved.
+
+Use of this source code is governed by the MIT license that can be found in the LICENSE file.
+*/
+
+use std::ops::{Mul, Sub};
+use std::ptr::NonNull;
+
+use crate::datum::time::USECS_PER_SEC;
+use crate::{direct_function_call, pg_sys, FromDatum, IntoDatum, PgBox};
+use pg_sys::{DAYS_PER_MONTH, SECS_PER_DAY};
+use pgx_sql_entity_graph::metadata::{
+    ArgumentError, Returns, ReturnsError, SqlMapping, SqlTranslatable,
+};
+
+#[cfg(feature = "time-crate")]
+const MONTH_DURATION: time::Duration = time::Duration::days(DAYS_PER_MONTH as i64);
+
+/// From the PG docs  https://www.postgresql.org/docs/current/datatype-datetime.html#DATATYPE-INTERVAL-INPUT
+/// Internally interval values are stored as months, days, and microseconds. This is done because the number of days in a month varies,
+/// and a day can have 23 or 25hours if a daylight savings time adjustment is involved. The months and days fields are integers while
+/// the microseconds field can store fractional seconds. Because intervals are usually created from constant strings or timestamp
+/// subtraction, this storage method works well in most cases...
+#[derive(Debug)]
+#[repr(transparent)]
+pub struct Interval(NonNull<pg_sys::Interval>);
+
+impl Interval {
+    /// This function takes `months`/`days`/`usecs` as input to convert directly to the internal PG storage struct `pg_sys::Interval`
+    /// - the sign of all units must be all matching in the positive or all matching in the negative direction
+    pub fn try_from_months_days_usecs(
```

Comment:
Per the other remark,
```suggestion
    pub fn try_from_months_days_micros(
```